dds/idl/marshal_generator.cpp

```diff
@@ -871,6 +871,10 @@ namespace {
       string field_name = fields[i]->local_name()->get_string();
       if (field_name == key_base) {
         AST_Type* field_type = fields[i]->field_type();
+        AST_Typedef* typedef_node = dynamic_cast<AST_Typedef*>(field_type);
```

Comment:
This could be `resolveActualType` from dds_generator.h.  That also handles typedefs of forward-declared (struct|union).